### PR TITLE
feat: opt out for vrm export improvements

### PR DESCRIPTION
**Pull request by AlejandroAlvarezMelucciDCL**
## What does this PR change?

This PR adds a Details Panel in the Backpack view that is available when the user has more than one wearable with the VRM block export feature turned on.
The Details Panel lets the user see a list of those items and allows for quick unequip/equip from that list and has a dynamic Export button that reacts to changes to equipped wearables.

...

## How to test the changes?

1. Pre-Requisite: Have wearables with the VRM block export enabled.
2. Launch the explorer
3. Equip one or more wearables that block the VRM export feature.
4. See that the VRM Export button on the left now shows a warning sign
5. After clicking said button, the Details Panel is shown and loads the info of the items that prevent the export.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

## Copilot summary

copilot:summary

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    {
+        public WearableDTO wearable;
+    }
+
+    [Serializable]
+    public struct WearableDTO
+    {
+        public string category;
+    }
+#endregion
```
Comment (sandrade-dcl) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs:132]: This is only a personal preference, but I would prefer to move these kind of structures to a separated file.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailItemComponentView.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        private bool isUnEquipAction;
+        private const string UNEQUIP_TEXT = "unequip";
+        private const string EQUIP_TEXT = "equip";
+
+        public event Action OnUnEquipWearable;
+        public event Action OnEquipWearable;
+
+        public void Start()
+        {
+            isUnEquipAction = true;
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailItemComponentView.cs:28]: Perhaps you can initialize the field on true, you could save this extra line

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsComponentView.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            foreach (var itemData in itemsToDisplay)
+            {
+                equippedItemsURNs.Add(itemData.urn);
+                var detailItem = wearableDetailPool.Get();
+                detailItem.gameObject.transform.SetParent(wearablesContainer, false);
+                pooledItems.Add(detailItem);
+
+                var detailComponentView = detailItem.gameObject.GetComponent<VRMDetailItemComponentView>();
+
+                if (detailComponentView)
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsComponentView.cs:98]: Is this case possible? If thats the case you will miss to show the item

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsComponentView.cs [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                        wearableUrn = itemData.urn,
+                        wearableImageUrl = itemData.thumbnail,
+                        wearableName = itemData.name,
+                        wearableCategoryName = itemData.data.wearable.category,
+                        wearableCreatorImageUrl = itemData.creatorImageUrl,
+                        wearableCreatorName = itemData.creatorName,
+                    };
+
+                    detailComponentView.SetModel(vrmItemModel);
+                    detailComponentView.ClearOnWearableUnequippedEvents();
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsComponentView.cs:111]: I think you should also clear the registered events when releasing the item to the pool. It saves undesired memory footprints

**Review by lorux0 — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                if (nftDataCache.TryGetValue(itemsToProcess[i], out NFTDataDTO itemData))
+                {
+                    itemsToDisplay.Add(itemData);
+                    itemsToProcess.RemoveAt(i);
+                }
+
+            // If we're still missing items, fetch them.
+            if (itemsToProcess.Count > 0)
+            {
+                string nftItemRawData = await WearablesFetchingHelper.GetNFTItems(itemsToProcess);
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs:67]: I think that making a static call here is not desirable. It makes impossible to properly make unit tests for the class

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    public class VRMDetailsController : IDisposable
+    {
+        private readonly IUserProfileBridge userProfileBridge;
+        private readonly IVRMDetailsComponentView view;
+        private readonly Dictionary<string, NFTDataDTO> nftDataCache = new ();
+
+        public event Action<string, UnequipWearableSource> OnWearableUnequipped;
+        public event Action<string, EquipWearableSource> OnWearableEquipped;
+
+        public VRMDetailsController(
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs:18]: We are missing unit tests for this and the rest of the added classes

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsComponentView.cs (hunk 3 above)
Comment (AlejandroAlvarezMelucciDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsComponentView.cs:98]: It shouldn't be possible, but I should also enforce it with a `[RequiresComponent]` tag in the prefab 👌 but regardless, good catch! That `if` was a leftover during testing this algorithm 💪

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsComponentView.cs (hunk 4 above)
Comment (AlejandroAlvarezMelucciDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsComponentView.cs:111]: Agreed, nice suggestion! I also took the opportunity to improve some more this method and the dispose/onDisable, thanks!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs (hunk 5 above)
Comment (AlejandroAlvarezMelucciDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs:67]: I get your point, but `WearablesFetchingHelper` isn't interfaced nor used in any other class that implements an interface I can use for injection in this controller and later mock in tests. I take you suggestion and note it as a tech debt for a later PR 👌

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs (hunk 6 above)
Comment (AlejandroAlvarezMelucciDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs:18]: I ran out of time to close this work in the cycle, I take you suggestion and note it as a tech debt for a later PR 👌

**Review by anicalbano — APPROVED** [review]
🟢 Reviewed by QA

https://github.com/decentraland/unity-renderer/assets/139157886/bcc6759c-8f1a-4f7a-9fb6-4e606e8d4e7b

**Review by sandrade-dcl — APPROVED** [review]
Looks good to me! Clean and clear, congrats! 🟢💃

**Pull request merged**